util/collate/collate_test.go

```diff
@@ -162,71 +147,70 @@ func (s *testCollateSuite) TestUnicodeCICollator(c *C) {
 		{"ﷻ", []byte{0x13, 0x5E, 0x13, 0xAB, 0x02, 0x09, 0x13, 0x5E, 0x13, 0xAB, 0x13, 0x50, 0x13, 0xAB, 0x13, 0xB7}},
 	}
 
-	testCompareTable(compareTable, "utf8mb4_unicode_ci", c)
-	testKeyTable(keyTable, "utf8mb4_unicode_ci", c)
+	testCompareTable(compareTable, "utf8mb4_unicode_ci", t)
+	testKeyTable(keyTable, "utf8mb4_unicode_ci", t)
 }
 
-func (s *testCollateSuite) TestSetNewCollateEnabled(c *C) {
+func TestSetNewCollateEnabled(t *testing.T) {
 	defer SetNewCollationEnabledForTest(false)
 
 	SetNewCollationEnabledForTest(true)
-	c.Assert(NewCollationEnabled(), Equals, true)
+	require.True(t, NewCollationEnabled())
 }
 
-func (s *testCollateSuite) TestRewriteAndRestoreCollationID(c *C) {
+func TestRewriteAndRestoreCollationID(t *testing.T) {
 	SetNewCollationEnabledForTest(true)
```

Comment:
Should we need to use `defer SetNewCollationEnabledForTest(false)` here?

Reply:
There is `SetNewCollationEnabledForTest(false)` later.

Reply:
So L177 is useless?

Reply:
I think it still makes sense before `SetNewCollationEnabledForTest(false)`. IMO, they are two groups of tests with new collation enabled/disabled.

Reply:
I think reset it to false seems better when the asserts L163-L166 are failed.